### PR TITLE
[develop] Replace existing UW CLI with UW API calls to template

### PR DESCRIPTION
**Pull request by elcarpenterNOAA**
## DESCRIPTION OF CHANGES: 

This PR continues the integration of the uwtools package by replacing current use of the UW CLI with UW API calls in Python scripts. These changes are limited to the UW template tool. 

### Type of change

- [x] New feature (non-breaking change which adds functionality)

## TESTS CONDUCTED: 

- [ ] hera.intel
- [ ] orion.intel
- [ ] hercules.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] derecho.intel
- [ ] gaea.intel
- [ ] gaeac5.intel
- [x] jet.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [x] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## DEPENDENCIES:

n/a

## DOCUMENTATION:

None. UW Documentation is up-to-date. 

## CHECKLIST

- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes do not require updates to the documentation (explain).
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

File: ush/create_aqm_rc_file.py [hunk 1]
```diff
@@ -124,36 +123,11 @@ def create_aqm_rc_file(cdate, run_dir, init_concentrations):
...
-            sys.exit(1)
-        finally:
-            print("Output:")
-            for line in output.split("\n"):
-                print(f"{indent * 2}{line}")
+    render(
+        input_file = Path(AQM_RC_TMPL_FP),
+        output_file = Path(aqm_rc_fp),
+        values_src = settings,
+    )
```
Comment (elcarpenterNOAA) [inline ush/create_aqm_rc_file.py]: These same simplifications are repeated throughout these 5 files. Further info on `template.render()` can be found on the uwtools [Read the Docs](https://uwtools.readthedocs.io/en/2.1.1/sections/user_guide/api/template.html).

**Review by christinaholtNOAA — APPROVED** [review]
LGTM!

**Review by mkavulich — APPROVED** [review]
Very happy to see removal of the ugly temp files and subprocesses! Looks good to me, assuming all tests pass.

**Review by MichaelLueken — APPROVED** [review]
@elcarpenterNOAA -

These changes look good to me!  I have also run fundamental tests on Derecho:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta_2  COMPLETE              19.75
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2_20240  COMPLETE              17.30
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              33.08
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR_2024042  COMPLETE              36.42
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0_20240425093  COMPLETE              33.99
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16_2024042509365  COMPLETE              48.73
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             189.27
```
and Hercules:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta_2  COMPLETE              11.11
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2_20240  COMPLETE               8.64
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              15.98
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR_2024042  COMPLETE             238.56
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0_20240425121  COMPLETE              77.96
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16_2024042512175  COMPLETE              26.14
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             378.39
```
and they all successfully passed.

Approving this PR now.

**Comment by MichaelLueken** [comment]
The Jenkins tests have successfully completed on Derecho, Gaea, Hera GNU, Hera Intel, and Hercules.  The Jet tests were manually run and all successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
community_20240425200459                                           COMPLETE              17.65
custom_ESGgrid_20240425200500                                      COMPLETE              24.92
custom_ESGgrid_Great_Lakes_snow_8km_20240425200501                 COMPLETE              20.71
custom_GFDLgrid_20240425200503                                     COMPLETE              10.07
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio_2021032018_202404  COMPLETE               8.07
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_netcdf_2022060112_48h_20  COMPLETE              88.64
get_from_HPSS_ics_RAP_lbcs_RAP_20240425200505                      COMPLETE              15.86
grid_RRFS_AK_3km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR_20240425200506  COMPLETE             619.54
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot_20  COMPLETE              64.13
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2_20240  COMPLETE               7.40
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta_2024  COMPLETE             919.70
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE            1796.69
```
Due to the continued degraded state of Orion and the additional maintenance today following the cooling issue that was encountered Wednesday night - Thursday morning, the Orion tests will not be run.

**Pull request merged**